soccer/src/soccer/strategy/agent/position/robot_factory_position.cpp

```diff
@@ -16,7 +15,7 @@ RobotFactoryPosition::RobotFactoryPosition(int r_id) : Position(r_id, "RobotFact
 std::optional<RobotIntent> RobotFactoryPosition::get_task(WorldState& world_state,
                                                           FieldDimensions& field_dimensions) {
     // If keeper, make no changes
-    if (robot_id_ == 0) {
+    if (robot_id_ == goalie_id_) {
```

Comment:
I will need to double-check with Sid about this, but if the goalie can change, we need to add functionality here to update the goalie if it is not assigned correctly.